src/Application/Actions/Donations/Confirm.php

```diff
@@ -92,9 +91,28 @@ protected function action(Request $request, Response $response, array $args): Re
             // Note that `on_behalf_of` is set up on create and is *not allowed* on update.
         ]);
 
-        $this->stripeClient->paymentIntents->confirm($paymentIntentId, [
-            'payment_method' => $pamentMethodId,
-        ]);
+        try {
+            $this->stripeClient->paymentIntents->confirm($paymentIntentId, [
+                'payment_method' => $pamentMethodId,
+            ]);
+        } catch (CardException $exception) {
+            $this->logger->info(sprintf(
+                'Stripe CardException on Confirm for donation %s (%s): %s',
+                $donation->getUuid(),
+                $paymentIntentId,
+                $exception->getMessage(),
+            ));
+
+            $this->entityManager->rollback();
+
+            return new JsonResponse([
+                'error' => [
+                    'message' => $exception->getMessage(),
+                    'code' => $exception->getStripeCode(),
+                    'decline_code' => $exception->getDeclineCode(),
+                ],
+            ], 402);
+        }
```

Comment:
Maybe worth also catching `\Stripe\Exception\ApiErrorException`, in case it wasn't a CardException and loggin details from that, particularly the result of `getStripeCode`

Reply:
Yep, good call. I think we probably still want this to be an `error` level log and a 500 code for the client – what do you reckon?

Reply:
Yes I think that's right, matchbot relies on stripe so if stripe is broken matchbot is broken and it should own up to being broken. Hopefully it'll never happen but if it does logs will help us.